app/src/ext_power_generic.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_ext_power_generic
+
+#include <device.h>
+#include <init.h>
+#include <drivers/gpio.h>
+#include <drivers/ext_power.h>
+
+#if DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT)
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+struct ext_power_generic_config {
+    const char *label;
+    const u8_t pin;
+    const u8_t flags;
+};
+
+struct ext_power_generic_data {
+    struct device *gpio;
+    bool status;
+};
+
+static int ext_power_generic_enable(struct device *dev) {
+    struct ext_power_generic_data *data = dev->driver_data;
+    const struct ext_power_generic_config *config = dev->config_info;
+
+    if (gpio_pin_set(data->gpio, config->pin, 1)) {
+        LOG_DBG("Failed to set ext-power control pin");
```

Comment:
This should be a warning, IMHO, not a debug:

```suggestion
        LOG_WRN("Failed to set ext-power control pin");
```